shared/src/main/scala/bloop/reporter/Problem.scala

```diff
@@ -13,12 +15,21 @@ final case class Problem private (
     /** Position in the source code where the message was triggered */
     position: xsbti.Position,
     /** The category of this problem. */
-    category: String
+    category: String,
+    /** Unique code attatched to the diagnostic being reported */
+    override val diagnosticCode: Optional[xsbti.DiagnosticCode]
```

Comment:
The `override val` here is odd but I believe it's needed because of the signature in Javaland. Is there a better way to do this?

Reply:
Looks good to me